$MirOS: ports/devel/cvs/patches/patch-src_commit_c,v 1.2 2007/02/01 23:29:35 tg Exp $
--- src/commit.c.orig	Sat Mar 10 23:14:20 2007
+++ src/commit.c	Sat Mar 10 23:14:14 2007
@@ -25,6 +25,8 @@
 #include "fileattr.h"
 #include "hardlink.h"
 
+__RCSID("$MirOS: src/gnu/usr.bin/cvs/src/commit.c,v 1.7 2007/02/01 23:24:27 tg Exp $");
+
 static Dtype check_direntproc (void *callerdat, const char *dir,
                                const char *repos, const char *update_dir,
                                List *entries);
@@ -613,31 +615,7 @@ commit (int argc, char **argv)
 
 	send_to_server ("ci\012", 0);
 	err = get_responses_and_close ();
-	if (err != 0 && use_editor && saved_message != NULL)
-	{
-	    /* If there was an error, don't nuke the user's carefully
-	       constructed prose.  This is something of a kludge; a better
-	       solution is probably more along the lines of #150 in TODO
-	       (doing a second up-to-date check before accepting the
-	       log message has also been suggested, but that seems kind of
-	       iffy because the real up-to-date check could still fail,
-	       another error could occur, &c.  Also, a second check would
-	       slow things down).  */
-
-	    char *fname;
-	    FILE *fp;
-
-	    fp = cvs_temp_file (&fname);
-	    if (fp == NULL)
-		error (1, 0, "cannot create temporary file %s", fname);
-	    if (fwrite (saved_message, 1, strlen (saved_message), fp)
-		!= strlen (saved_message))
-		error (1, errno, "cannot write temporary file %s", fname);
-	    if (fclose (fp) < 0)
-		error (0, errno, "cannot close temporary file %s", fname);
-	    error (0, 0, "saving log message in %s", fname);
-	    free (fname);
-	}
+	logmsg_cleanup(err);
 	return err;
     }
 #endif
@@ -702,6 +680,7 @@ commit (int argc, char **argv)
 	sleep_past (last_register_time);
     }
 
+    logmsg_cleanup(err);
     return err;
 }
 
@@ -853,7 +832,7 @@ check_fileproc (void *callerdat, struct 
 	case T_ADDED:
 	case T_REMOVED:
         {
-            char *editor;
+            char *editor = NULL;
 
 	    /*
 	     * some quick sanity checks; if no numeric -r option specified:
@@ -1250,6 +1229,7 @@ precommit_proc (const char *repository, 
 #endif /* SUPPORT_OLD_INFO_FMT_STRINGS */
 			      filter,
 			      "c", "s", cvs_cmd_name,
+			      "I", "s", global_session_id,
 #ifdef SERVER_SUPPORT
 			      "R", "s", referrer ? referrer->original : "NONE",
 #endif /* SERVER_SUPPORT */
@@ -2255,12 +2235,13 @@ checkaddfile (const char *file, const ch
 		   this was added into the log message. */
 		t = time (NULL);
 		ct = gmtime (&t);
-		tmp = Xasprintf ("file %s was added on branch %s on %d-%02d-%02d %02d:%02d:%02d +0000",
+		tmp = Xasprintf ("file %s was added on branch %s on %ld-%02d-%02d %02d:%02d:%02d +0000",
 				 file, tag,
-				 ct->tm_year + (ct->tm_year < 100 ? 0 : 1900),
+				 (long)ct->tm_year
+				  + (ct->tm_year < 100 ? 0 : 1900),
 				 ct->tm_mon + 1, ct->tm_mday,
 				 ct->tm_hour, ct->tm_min, ct->tm_sec);
-			 
+
 		/* commit a dead revision. */
 		revnum = RCS_whatbranch (rcs, tag);
 		retcode = RCS_checkin (rcs, NULL, NULL, tmp, revnum, headtime,
